public/pages/utils/anomalyResultUtils.ts

```diff
@@ -992,64 +948,67 @@ export const getTopAnomalousEntitiesQuery = (
   detectorId: string,
   size: number,
   sortType: AnomalyHeatmapSortType,
+  isMultiCategory: boolean,
   isHistorical?: boolean,
   taskId?: string
 ) => {
   const termField =
     isHistorical && taskId ? { task_id: taskId } : { detector_id: detectorId };
 
-  const requestBody = {
-    size: 0,
-    query: {
-      bool: {
-        filter: [
-          {
-            range: {
-              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
-                gt: 0,
+  // To handle BWC, we will return 2 possible queries based on the # of categorical fields:
+  // (1) legacy way (1 category field): bucket aggregate over the single, nested, 'entity.value' field
+  // (2) new way (>= 2 category fields): bucket aggregate over the new 'model_id' field
+  const requestBody = isMultiCategory
+    ? {
+        size: 0,
+        query: {
+          bool: {
+            filter: [
+              {
+                range: {
+                  [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                    gt: 0,
+                  },
+                },
               },
-            },
-          },
-          {
-            range: {
-              data_end_time: {
-                gte: startTime,
-                lte: endTime,
+              {
+                range: {
+                  data_end_time: {
+                    gte: startTime,
+                    lte: endTime,
+                  },
+                },
               },
-            },
-          },
-          {
-            term: termField,
+              {
+                term: termField,
+              },
+            ],
           },
-        ],
-      },
-    },
-    aggs: {
-      [TOP_ENTITIES_FIELD]: {
-        nested: {
-          path: ENTITY_FIELD,
         },
         aggs: {
           [TOP_ENTITY_AGGS]: {
             terms: {
-              field: ENTITY_VALUE_PATH_FIELD,
+              field: MODEL_ID_FIELD,
```

Comment:
Would this work for historical detector?  I don't recall @ylwu-amzn  has added model id field.

Reply:
I believe it does. I've tested on the test domain and historical multi-category results are returning model IDs. @ylwu-amzn can you confirm if this is part of what's included in [this PR](https://github.com/opensearch-project/anomaly-detection/pull/159)?

Reply:
just checked.  Seems Yaliang did add model id: https://github.com/opensearch-project/anomaly-detection/blob/f0e265e5f4ed45f5af491a7a43a17b735580289b/src/main/java/org/opensearch/ad/task/ADBatchTaskRunner.java#L1082